File: IAIso-v5.0/core/iaiso-ruby/lib/iaiso/core/step_outcome.rb
# frozen_string_literal: true

module IAIso
  module Core
    # Outcome of a single PressureEngine#step call. Wire form = string value.
    module StepOutcome
      OK         = "ok"
      ESCALATED  = "escalated"
      RELEASED   = "released"
      LOCKED     = "locked"

      ALL = [OK, ESCALATED, RELEASED, LOCKED].freeze
    end
  end
end

File: IAIso-v5.0/core/iaiso-php/src/Core/StepOutcome.php
<?php

declare(strict_types=1);

namespace IAIso\Core;

/** Outcome of a {@see PressureEngine::step()} call. Wire form is lowercase. */
enum StepOutcome: string
{
    case Ok         = 'ok';
    case Escalated  = 'escalated';
    case Released   = 'released';
    case Locked     = 'locked';
}
